src/testModel.ts

```diff
@@ -225,25 +224,33 @@ export class TestModel {
     if (allFilesLoaded)
       return [];
     const { rootSuite, errors } = await this._playwrightTest.listTests(this.config, files);
-    this._updateProjects(rootSuite.suites, files);
-    return errors;
+    this._updateProjects(rootSuite.suites, files, errors);
   }
 
-  async listTests(files: string[]): Promise<reporterTypes.TestError[]> {
+  async listTests(files: string[]) {
     const { rootSuite, errors } = await this._playwrightTest.listTests(this.config, files);
-    this._updateProjects(rootSuite.suites, files);
-    return errors;
+    this._updateProjects(rootSuite.suites, files, errors);
   }
 
-  private _updateProjects(newProjectSuites: reporterTypes.Suite[], requestedFiles: string[]) {
+  private _updateProjects(newProjectSuites: reporterTypes.Suite[], requestedFiles: string[], errors: reporterTypes.TestError[]) {
+    for (const requestedFile of requestedFiles)
+      this._errors.deleteAll(requestedFile);
+    for (const error of errors) {
+      if (error.location)
+        this._errors.set(error.location.file, error);
+    }
+
     for (const [projectName, project] of this._projects) {
       const files = projectFiles(project);
       const newProjectSuite = newProjectSuites.find(e => e.project()!.name === projectName);
       const filesToClear = new Set(requestedFiles);
       for (const fileSuite of newProjectSuite?.suites || []) {
+        if (this._errors.has(fileSuite.location!.file))
```

Comment:
I wonder how can we get a non-empty suite for a file that has a loading error?

Reply:
that would be a subsequent load upon watch